_episodes/01-introduction.md

```diff
@@ -3,13 +3,247 @@ title: "Introduction"
+
+
+### 5. Choose a loss function and metrics
+
+The loss function tells the training algorithm how far away the predicted value was from the true value. There are a number of different types of function we can use here that might be suited to different types of data and network architectures. We'll look at choosing a loss function in more detail later on.
+
+
+### 6. Train the model
+
+Before we can train the model we have one final component to decide upon, the optimizer. The optimizer is responsible for taking the output of the loss function and then applying some changes to weights within the network. We can think of the training process as like trying to gradually descend a (very jagged looking) curve on a graph (that looks more like a rocky mountain side than a smooth curve) and find the bottom of it, without knowing in advance where that bottom is. Along the way we might hit some flat or even rising parts of the curve that make it look like we've reached the bottom when we haven't, we call these local minima. Different optimizers and optimizer parameters can help prevent us getting stuck in local minima. As with the loss function the best choice may depend on the kind of data and network architecture that we're using.
+
+Now that we've set the optimizer function we can go ahead and start training our network. We'll probably keep doing this for a given number of iterations or epochs or until the loss function gives a value under a certain threshold.
+
+[//]: # "is a graph of a reduction in loss sensible here?"
+
+### 7. Tune Hyperparameters
+
+Hyperparameters are all the parameters set by the person configuring the machine learning instead of those learned by the algorithm itself. The Hyperparameters include the number of epochs or the parameters for the optimizer. If we are really trying to optimize things then we might even do a sweep through a whole range of Hyperparameters and use the loss function to decide which Hyperparameters performed best.
+
+### 8. Measure Performance
+
+Once we think the network is performing well we want to measure its performance. To do this we might use the validation set of data that we put aside earlier and didn't use as part of the training process. For a classification task we can often evaluate four different measures for each possible class in the data:
+
+1. True Positives: Where we correctly classified something as belonging to this class.
+2. True Negatives: Where we correctly classified that something wasn't a member of this class.
+3. False Positives: Where we incorrectly classified something as belonging to this class when it doesn't.
+4. False Negatives: Where we incorrectly classified something as not belonging to this class when it does.
```

Comment:
I would leave this out, it is fairly specific for classification and is easier to explain with an example. Maybe just mention in general that there are several ways of computing one score that tells us how well our model performs (such as accuracy or F1 for classification, RMSE for regression) and that the choice of this metric depends on the context of the model

Reply:
I think we should probably just say something like there are many ways to achieve this and we'll discuss the details of them later.